frontend/src/api.ts

```diff
@@ -1,6 +1,34 @@
 import axios from 'axios';
 
-const API_URLS = {
+export interface Post {
+    title: string,
+    content: string,
+    datetime: string,
+    hash: string,  // TODO it should be a hexstring with 0x prefix
+}
+
+export interface Comment {
+    content: string,
+    datetime: string,
+    hash: string,
+    post: string,  // TODO it actually has to be a valid URL to a post
+}
+
+export interface Vote {
+    hash: string,
+    datetime: string,
+    isUpvote: boolean,
+}
+
+export interface PostVote extends Vote {
```

Comment:
this is a bit confusing, why not add Id to Post?

Reply:
That's kinda tricky (I didn't know how to make it simpler):

Objects right here are created *before* they are sent to the API and the API assigns the IDs and all other stuff (for example you ma look how `ApiPost` looks like - it's the one we fetch from the API after successful submission).

The interfaces right here are data required by the API to create a new entry in the database, and the ID is not known yet, so I can't add it to `Post`.

I'd like to make it more clear that those are not the *real* types used later in the app (maybe rename them to `PostData` etc.) but I'm not sure how to do it best.

The API expects that a new Vote contains a number referencing the Post/Comment that the vote targets so I pass it with the `postId` / `commentId` which are extracted from `ApiPost` or `ApiComment`.

Btw another annoying thing with the API is that adding a comment to a post requires the post's URL instead of ID which is inconsistent. I'm planning to work on improving all that after redux-saga (I'll rework the API itself)